design-documents/frontend/admin-guidelines.md

```diff
@@ -0,0 +1,194 @@
+# Magento Admin Guidelines
+
+## Problem
+
+There are already a handful of React applications built for Magento Admin use, but they have different dependencies and versions between them and we don't yet have a micro front-end architecture to manage them.
+
+## Solution (for now)
+
+We must do our best to align common dependencies manually.
+
+The following [`package.json`](#packagejson) includes a list of dependencies that you should use instead of their alternatives. Please also stick within the version ranges specified to support a graceful transition to a future architecture.
+
+To aid in the process of using these dependencies, consider using the [TSX Yeoman Generator](https://www.npmjs.com/package/generator-tsx).
+
+For more information behind these technology choices, refer to the [Technical Vision](./technical-vision.md).
+
+## package.json
+
+Each section of the `package.json` file has been broken down into different topics.
+
+- [version](#version)
+- [scripts](#scripts)
+- [browserslist](#browserslist)
+- [dependencies](#dependencies)
+  - [Utility functions](#utility-functions)
+  - [SSR](#ssr)
+  - [React](#react)
+  - [State management](#state-management)
+  - [Code splitting](#code-splitting)
+  - [i18n](#i18n)
+  - [Theming](#theming)
+  - [Relay](#relay)
+  - [TypeScript import helpers](#typescript-import-helpers)
+- [devDependencies](#devdependencies)
+  - [@types](#types)
+  - [Redux mock store](#redux-mock-store)
+  - [Testing library](#testing-library)
+
+### version
+
+The version of this `package.json` follows [semver](https://semver.org/) and will change with this document accordingly.
+
+```json
+"version": "0.1.0",
+```
+
+### scripts
+
+Customize your CRA configuration with [craco][].
+
+- [Don't eject](https://create-react-app.dev/docs/alternatives-to-ejecting).
+- Avoid forking, because you can oftentimes do the same thing by creating your own [craco][] plugin. See [craco plugins on npm](https://www.npmjs.com/search?q=craco%20plugin) and the [craco recipes](https://github.com/sharegate/craco/tree/master/recipes) for guidance.
+
+```json
+"scripts": {
+  "start": "craco start",
+  "build": "craco build",
+  "test": "craco test"
+},
+```
+
+### [browserslist](https://github.com/browserslist/browserslist#readme)
+
+Don't forget your [polyfills](https://create-react-app.dev/docs/supported-browsers-features#supported-language-features)!
+
+```json
+"browserslist": [
+  ">0.2%",
+  "not dead",
+  "not ie <= 11",
+  "not op_mini all"
+],
+```
+
+### dependencies
+
+#### Utility functions
+
+Use vanilla (native) JS if possible, utility functions if not. [You don't (may not) need Lodash/Underscore](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore#you-dont-may-not-need-lodashunderscore-). See also [just](https://github.com/angus-c/just#just).
```

Comment:
I can see how you need some clarification here.

1. We should not be loading the entire Lodash library on the client, which I think is obvious for many years now.
1. You shouldn't be installing utility functions, Lodash or otherwise, if you can get the same functionality in vanilla JS.
1. If you truly need some utility functions, it's probably only a handful of them. I don't think it makes sense to hoist these dependencies in a Micro FE environment, should that ever be introduced, so you can install whatever you want. My personal preference is to install something with TypeScript support.
   - Lodash has 3rd-party TS support.
   - [Just](https://github.com/angus-c/just#just) has "some" 3rd-party contributions, but [never plans to](https://github.com/angus-c/just/issues/126#issuecomment-470807747) support it officially.
   - [Queso](http://queso.surge.sh/) has 1st-class TS support and documentation, but is a new project of my own making (on paternity leave). I haven't even announced it, but it's super easy to contribute to and fully tested.

Reply:
Addressed in https://github.com/magento/architecture/pull/231/commits/b1c19bc24ff6a13764b7aa3d7e9a2c64daaebb16